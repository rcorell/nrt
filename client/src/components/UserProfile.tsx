import { useQuery } from '@apollo/client';
import React from 'react';

import { FetchUserQuery } from 'src/api/__generated__/FetchUserQuery';
import { fetchUserQuery } from 'src/api/api';
import { LOADING_TEXT } from 'src/components/shared';
import { setBrowserTitle } from 'src/utils';

export const UserProfile: React.FC = () => {
    const query = useQuery<FetchUserQuery>(fetchUserQuery);

    setBrowserTitle('User Profile');

    if (query.loading) {
        return <b>{LOADING_TEXT}</b>;
    }

    if (query.error) {
        return <div>{JSON.stringify(query.error)}</div>;
    }

    return <h2>{query.data!.user.name}</h2>;
};
