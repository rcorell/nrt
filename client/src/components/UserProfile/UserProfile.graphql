query FetchUser {
    user {
        id
        name        
    }
}