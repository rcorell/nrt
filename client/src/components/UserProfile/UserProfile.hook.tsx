import { useFetchUserQuery } from 'src/api/__generated__/types';

export const useUser = () => {
    const { data, error, loading, refetch } = useFetchUserQuery({
        fetchPolicy: 'network-only'
    });

    return { error, loading, refetch, user: data?.user };
};
