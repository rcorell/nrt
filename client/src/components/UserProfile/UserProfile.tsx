import React from 'react';
import { LOADING_TEXT } from 'src/components/shared';
import { setBrowserTitle } from 'src/utils';

import { useUser } from './UserProfile.hook';

export const UserProfile: React.FC = () => {
    const userHook = useUser();

    setBrowserTitle('User Profile');

    if (userHook.loading) {
        return <b>{LOADING_TEXT}</b>;
    }

    if (userHook.error) {
        return <div>{JSON.stringify(userHook.error)}</div>;
    }

    return <h2>{userHook.user!.name}</h2>;
};

UserProfile.displayName = 'UserProfile';
