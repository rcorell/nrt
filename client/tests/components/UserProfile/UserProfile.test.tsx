import { UserProfile } from 'src/components/UserProfile/UserProfile';

import { createComponentMocks, testComponent } from 'tests/testHelpers';

const mocks = createComponentMocks({
    user: {
        name: 'Buckaroo Banzai'
    }
});

jest.mock('src/components/UserProfile/UserProfile.hook', () => ({
    useUser: () => mocks[mocks.mockState]
}));

testComponent(UserProfile, mocks);
