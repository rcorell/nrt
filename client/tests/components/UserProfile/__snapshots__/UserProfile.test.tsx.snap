// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile failure 1`] = `
<div>
  <div>
    "Error"
  </div>
</div>
`;

exports[`UserProfile loading 1`] = `
<div>
  <b>
    Loading...
  </b>
</div>
`;

exports[`UserProfile success 1`] = `
<div>
  <h2>
    Buckaroo Banzai
  </h2>
</div>
`;
